components/VotePanel/hooks.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  useAddressQuery_CommunityTokenOwner,
+  useAddressQuery_CouncilTokenOwner,
+} from '@hooks/queries/addresses/tokenOwner'
+import { useAddressQuery_SelectedProposalVoteRecord } from '@hooks/queries/addresses/voteRecord'
+import { useVoteRecordByPubkeyQuery } from '@hooks/queries/voteRecord'
+import { useHasVoteTimeExpired } from '@hooks/useHasVoteTimeExpired'
+import useRealm from '@hooks/useRealm'
+import { ProposalState, GoverningTokenRole } from '@solana/spl-governance'
+import useWalletStore from 'stores/useWalletStore'
+
+export const useIsVoting = () => {
+  const { governance, proposal } = useWalletStore((s) => s.selectedProposal)
+  const hasVoteTimeExpired = useHasVoteTimeExpired(governance, proposal!)
+
+  const isVoting =
+    proposal?.account.state === ProposalState.Voting && !hasVoteTimeExpired
+  return isVoting
+}
+
+export const useVotingPop = () => {
+  const { tokenRole } = useWalletStore((s) => s.selectedProposal)
+
+  const votingPop =
+    tokenRole === GoverningTokenRole.Community ? 'community' : 'council'
+
+  return votingPop
+}
+
+export const useVoterTokenRecord = () => {
+  const { tokenRole } = useWalletStore((s) => s.selectedProposal)
+  const { ownTokenRecord, ownCouncilTokenRecord } = useRealm()
+
+  const voterTokenRecord =
+    tokenRole === GoverningTokenRole.Community
+      ? ownTokenRecord
+      : ownCouncilTokenRecord
+  return voterTokenRecord
+}
+
+export const useProposalVoteRecordQuery = (quorum: 'electoral' | 'veto') => {
```

Comment:
Here I have rewritten our code for fetching user `VoteRecord`s using react-query. It is:
- self-contained
- faster than previous (useWalletStore actually uses a getProgramAccounts RPC call and then filters it down to one account for some reason which is way way slower)
- much more complete API 

I do not migrate the entire codebase to use this new hook.